release-notes.txt

```diff
@@ -1,3 +1,565 @@
+
+CPU
+---
+
++ core: add stdio.h to replace stdout functions with stdio_null (#20872)
+* cpu/native: Gardening/QoL (#21283)
+* cpu/nrf5*: mark and resolve radio conflict (#21250)
+* cpu/nrf52/radio/nrf802154: fix beacon acceptance (#20982)
+* cpu/sam0_common/flashpage: disable interrupts while writing (#21326)
+* cpu/sam0_eth: fix hang with broken PHY (#21172)
+* cpu/sam3: fix hwrng peripheral register access (#21177)
+* cpu/sam3: optimize gpio ISR processing (#21179)
+* cpu/samd5x: improve can-initialization (#21173)
+* cpu/stm32{f0,g0,c0}: fix ADC initialization sequence (#21230)
+* dist/tools/esptools: Upgrade the ESP32 toolchain to GCC v14.2, GDB
+  v14.2 and OpenOCD v0.12 (#21144)
+* {cpu/sam0_common/sam0_sdhc,drivers/mtd}: do not allocate `work_area`
+  twice (#21169)
+
+Device Drivers
+--------------
+
++ drivers/periph_timer: add `timer_get_closest_freq()` (#20581)
++ drivers/sen5x: Add device driver for SEN5x (#19955)
++ drivers/tm1637: add driver and tests for tm1637 7-segment display (#21112)
+* driver/sen5x: Small Fixups (#21392)
+* drivers/at: at_readline_stop_at_str fix remaining buffer length (#21195)
+* drivers/sdmmc/sdmmc_sdhc: remove DMA check (#21260)
+* drivers/{mtd_sdmmc,sdmmc}: reinit card (#21305)
+
+Documentation
+-------------
+
++ add *.doc.md to doxygen sources (#21319)
++ boards/nrf52840{dk,dongle}: add MCU table, move to doc.md (#21271)
++ boards/nucleo-f413zh: add MCU table, move to doc.md (#21264)
++ dist/tools: Add Doxygen as a Tool for CI (#21300)
++ doc, buildsystem: Add a Version Check for Doxygen, Fix doc* Target
+  Execution (#21277)
++ doc/managing-a-release: add looking out for release tests to preps (#21368)
++ doc: add documentation on RIOT_FILE_RELATIVE (#21170)
++ doc: add psutil requirement to  Getting Started (#21399)
+* boards/arduino-due: update doc.txt (#21207)
+* boards/nucleo-144: rename doc.txt to doc.md (#21318)
+* boards/nucleo-f746zg: board doc update (#21310)
+* boards/rpi-pico*: Update and fix documentation and image (#21218)
+* build-system: Drop -Wno-error=documentation (#21377)
+* doc/porting-boards: Change example doc.txt to doc.md (#21257)
+* doc: doc.txt -> doc.md (#21255)
+* doc: hack to hide doc.md files until doxygen gets fixed (#21273)
+* doc: remove and replace deprecated mailing lists (#21285)
+* doc: Use the correct Doxygen Logo File Type in the Footer (#21274)
+* docs/rust: Install c2rust with --locked (#21236)
+* docs: Switch out pkg reference link (#21229)
+* sys/cord/doc.txt: Update additional references to RFC 9176 (#21159)
+* sys/malloc_monitor: Fix documentation (#21225)
+* sys/psa_crypto/doc: correctly end code block (#21380)
+* treewide: fix double the in doc and comments (#21224)
+* treewide: fix example references in docs (#21212)
+* CODING_CONVENTION.md: migrate to `#pragma once` (#21369)
+
+Build System / Tooling
+----------------------
+
++ buildsystem: Enable -Wdocumentation for LLVM (#21358)
++ dist/tools: add `#pragma once` to headerguard check (#21367)
```

Comment:
This might also be something that we might want to point out to RIOT users that new code should use `#pragma once` instead of classic include guards.